src/VisualStudio/Core/Test.Next/Services/SolutionServiceTests.cs

```diff
@@ -191,42 +191,110 @@ public class SolutionServiceTests
+                var projectId = workspace.CurrentSolution.ProjectIds.First();
+                var additionalDocumentId = DocumentId.CreateNewId(projectId);
+                var additionalDocumentInfo = DocumentInfo.Create(
+                    additionalDocumentId, "additionalFile",
+                    loader: TextLoader.From(TextAndVersion.Create(SourceText.From("test"), VersionStamp.Create())));
+
+                await VerifySolutionUpdate(workspace, s =>
+                {
+                    return s.AddAdditionalDocument(additionalDocumentInfo);
+                });
+
+                workspace.OnAdditionalDocumentAdded(additionalDocumentInfo);
+
+                await VerifySolutionUpdate(workspace, s =>
+                {
+                    return s.WithAdditionalDocumentText(additionalDocumentId, SourceText.From("changed"));
+                });
+
+                await VerifySolutionUpdate(workspace, s =>
+                {
+                    return s.RemoveAdditionalDocument(additionalDocumentId);
+                });
+            }
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.RemoteHost)]
+        public async Task TestDocument()
+        {
+            var code = @"class Test { void Method() { } }";
+
+            using (var workspace = await TestWorkspace.CreateCSharpAsync(code))
+            {
+                var projectId = workspace.CurrentSolution.ProjectIds.First();
+                var documentId = DocumentId.CreateNewId(projectId);
+                var documentInfo = DocumentInfo.Create(
+                    documentId, "sourceFile",
+                    loader: TextLoader.From(TextAndVersion.Create(SourceText.From("class A { }"), VersionStamp.Create())));
+
+                await VerifySolutionUpdate(workspace, s =>
+                {
+                    return s.AddDocument(documentInfo);
+                });
+
+                workspace.OnDocumentAdded(documentInfo);
+
+                await VerifySolutionUpdate(workspace, s =>
+                {
+                    return s.WithDocumentText(documentId, SourceText.From("class Changed { }"));
+                });
+
+                await VerifySolutionUpdate(workspace, s =>
+                {
+                    return s.RemoveDocument(documentId);
+                });
+            }
+        }
+
         private static async Task VerifySolutionUpdate(string code, Func<Solution, Solution> newSolutionGetter)
         {
             using (var workspace = await TestWorkspace.CreateCSharpAsync(code))
             {
-                var map = new Dictionary<Checksum, object>();
+                await VerifySolutionUpdate(workspace, newSolutionGetter);
+            }
+        }
 
-                var solution = workspace.CurrentSolution;
-                var service = await GetSolutionServiceAsync(solution, map);
+        private static async Task VerifySolutionUpdate(TestWorkspace workspace, Func<Solution, Solution> newSolutionGetter)
```

Comment:
this is just extract method. there is actually no change here. not sure why github think there is change here.